libs/librepcb/common/cam/excellongenerator.cpp

```diff
@@ -47,7 +48,38 @@ ExcellonGenerator::~ExcellonGenerator() noexcept {
 
 void ExcellonGenerator::drill(const Point&          pos,
                               const PositiveLength& dia) noexcept {
-  mDrillList.insert(*dia, pos);
+  mDrillList.insert(dia, pos);
+  if (!mToolList.contains(dia))
+    mToolList.append(dia);
+}
+
+void ExcellonGenerator::slot(const Point&          pos,
+                             const PositiveLength& width,
+                             const PositiveLength& height,
+                             const Angle&          rotation) noexcept {
+  if (height < width)
+    slot(pos, height, width, rotation + Angle::deg90());
+  else if (height == width)
+    drill(pos, width);
+  else {
+    if (!mToolList.contains(width))
+      mToolList.append(width);
+    Point start = pos;
+    Point end = pos;
+
+    start.setY(start.getY()-height/2+width/2);
+    end.setY(start.getY()+height-width);
+
+    start.rotate(rotation, pos);
+    end.rotate(rotation, pos);
+
+    struct slotData temp = {
+      .start = start,
+      .end = end,
+      .width = width
+    };
+    mSlotList.append(temp);
+  }
```

Comment:
Yeah I wasn't quite sure where to put the conversion, but I figured that since the diameter+startpos+endpos representation is Excellon-specific, it would make sense to keep it local to Excellon-specific code, to keep the code as modular as possible, but putting the conversion logic in BoardGerberExport is also an option.

AFAIK Excellon slots with identical start- and endpoint are defined to be equivalent to drills, so it shouldn't matter, but it might indeed be cleaner to just create a slot anyhow

Reply:
I moved the slot representation conversion logic into the BoardGerberExport class